Adding embedded tweets

ercadev opened the pull request:
Have done some rough work on this. Added example if someone is interested in trying it out.
I need to do some refactoring and adding documentation to the readme.

Thoughts on the building of the tweet like: 
```
TweetEmbed.fromTweet(
              Tweet.fromRawJson(
                snapshot.data,
              ),
              backgroundColor: (darkMode) ? Colors.grey[800] :Colors.white,
              darkMode: darkMode,
              createdDateDisplayFormat: DateFormat("EEE, MMM d, ''yy"),
            )
```
The thought is that you should be able to choose between light or dark textTheme and then specify the backgroundColor so it matches your app.

kawa89 commented:
I will check it on Monday ;-) before I can merge it please add instructions to to the readme file ;-)

kawa89 commented on example/lib/EmbeddedTweet.dart:
```diff
@@ -0,0 +1,104 @@
+
+/// Widget containing 4 Tweet types:
+/// TweetView, CompactTweetView, TweetView with a quoted Tweet, CompactTweetView with a quoted Tweet
+class EmbeddedTweetPage extends StatefulWidget {
```
Change filename so it will be in snake_case

kawa89 commented on example/lib/tweet_page.dart:
```diff
@@ -55,6 +62,39 @@ class TweetPage extends StatelessWidget {
+              Tweet.fromRawJson(
+                snapshot.data,
+              ),
+              backgroundColor: Colors.grey[800],
```
you have the dark mode in the embedded section you added to the main screen to the example. Here it would be great to show light versions of tweets and change the background to white (remove line 29 in this file)

kawa89 commented on example/lib/tweet_page.dart:
```diff
@@ -26,11 +26,18 @@ class TweetPage extends StatelessWidget {
+          Divider(height: 80,),
+          buildHeader("$mediaType Quote TweetEmbed"),
+          buildTweetEmbed(quoteTweetPath),
+          Divider(height: 80,),
```
Maybe smaller dividers? And add the dividers to all sections or tweets, to be consistent.

kawa89 commented on example/lib/tweet_page.dart:
```diff
@@ -26,11 +26,18 @@ class TweetPage extends StatelessWidget {
         children: <Widget>[
+          buildHeader("$mediaType TweetEmbed"),
+          buildTweetEmbed(tweetPath),
+          Divider(height: 80,),
```
format the file. Format all the files ;)

kawa89 commented on example/lib/tweet_page.dart:
```diff
@@ -26,11 +26,18 @@ class TweetPage extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return Scaffold(
+      backgroundColor: Colors.grey[900],
```
the background should be by default white.

kawa89 commented on lib/src/profile_image_embedded.dart:
```diff
@@ -0,0 +1,28 @@
+      placeholder: (context, url) => Container(height: 40, width: 40),
+      errorWidget: (context, url, error) => Icon(Icons.error),
+    );
+    */
```
remove unused/commented code

kawa89 commented on lib/src/tweet_text_embedded.dart:
```diff
@@ -0,0 +1,116 @@
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      margin: EdgeInsets.only(top: 15,bottom: 15),
```
This is a line that differs from tweet_text.dart - try to reuse that widget.

kawa89 commented on lib/src/twitter_logo_embedded.dart:
```diff
@@ -0,0 +1,18 @@
+import 'package:flutter/material.dart';
+
+class TwitterLogo extends StatelessWidget {
```
try to reuse twitter_logo.dart, for example pass the size via constructor

kawa89 commented on lib/src/byline_quote_embedded.dart:
```diff
@@ -0,0 +1,97 @@
+import 'package:tweet_ui/src/view_mode.dart';
+
+/// Widget that displays user name, user screen name (the @ name), if the user is verified
+class BylineQuote extends StatelessWidget {
```
Try to reuse the original byline file. Maybe pass a child widget with this:

```
                  Padding(
                    padding: const EdgeInsets.symmetric(horizontal: 4.0),
                    child: Text(
                      "• " + tweetVM.createdAt,
                      style: userScreenNameStyle,
                      textAlign: TextAlign.start,
                      softWrap: false,
                    ),
                  ),
```

kawa89 commented:
@ercadev will you fix this in the nearest future?

ercadev commented:
> @ercadev will you fix this in the nearest future?

Yes, sorry. Have been busy. Will try fix it tomorrow!

kawa89 commented:
Nice ;) no hurries, just wanted to know the status on this and if I should reserve some time to fix this ;)
Thanks for a quick response :-)

ercadev commented:
So I've worked on the comments you made! Some stuff that you suggested would use original widget Instead of a widget_embedded.dart I just removed or i added som parameters.

About the readme... Would it be okay to start a new section like "Embedded tweets"? I feel like i would be cluttering to add more screen shots into the table right now. Do you have any preffered way?

kawa89 commented:
Yeah, I know it's cluttered :( You can rearrange the readme. 
Maybe we should use a wiki for the details to keep the readme as simple as possible?

ercadev commented:
Been thinking about the readme. I feel like the section "Styling" maybe could move to a wiki. I think the screenshots and getting started should stay. But maybe I should just squeeze my stuff in the readme now and we can open an issue for rewriting the readme. How does that sound?

kawa89 commented:
👍 We can do that. I've planned some time on Friday for this repo, so I can change the readme/wiki then ;)
